content/articles/graph-cycle-detection-c#/index.md

```diff
@@ -0,0 +1,431 @@
+A graph is a non-linear data structure that consists of:-
+
+1. Some grouped vertices are also called nodes.
+
+2. Some ordered pairs of the form (u, v) are called edges.
+
+A graph is like a tree but without any cycles. We do not have a root node in graphs. Below, is an example of a graph with four nodes or vertex and six edges or lines. We don't have a limitation of how many connections we can have from one node. If two nodes are connected, we say they are adjacent or neighbors. `John` and `bob` are neighbors. `John` and `Sam` are not because they are not connected.
+
+[image of a Graph](/engineering-education/graph-cycle-detection/graphexample.png/)
+
+If the edges have a direction, we say we have a directed graph. From the directed graph image below, `John` is connected to `Bob` but the opposite is not true. This is like how Twitter works. If you follow someone, there is a connection from your account to their account. Not the other way round unless they follow you as well.
+
+[image of a directed-graph](/engineering-education/graph-cycle-detection/directed-graph.png/)
+
+There are undirected graphs as well. An example of this can be Facebook. When you add a friend, there is a connection from you to them. The Vice verse is true. The connections do not have a direction.
+
+The edges can also have weights. We use these weights to represent how strong the connection is, For example, Facebook. When two people communicate, we can put more weight on them. We then use that to show their best friends using the nodes that have the highest weights.
+
+[Image of Weighted Graph](/engineering-education/graph-cycle-detection/weightedgraph.png/)
+
+With this brief description, we can get to know what a cycle is and how to detect that in a graph.
+
+#### Prerequisite
+
+To be able to follow this article well, one needs:
+
+1. To have [Visual Studio](https://visualstudio.microsoft.com/) installed.
+
+2. Have an understanding of Recursion. [Recursion](https://www.youtube.com/watch?v=IJDJ0kBx2LM&t=163s)
+
+3. Some understanding of how to build a graph using an adjacency list and matrix.
+
+   -[Adjacency list](https://www.section.io/engineering-education/graphs-in-data-structure-using-cplusplus/)
+
+#### Key Takeaways 
+
+- Understand What is a Cycle in a Graph.
+
+- Understand How to detect a Cycle. 
+
+      1) Depth-first search algorithm.
+
+            - Cycle Detection on a directed graph
+
+            - Cycle Detection on an undirected graph
```

Comment:
Combining both uppercase and lowercase on a sentence? Combining both uppercase and lowercase on a sentence?